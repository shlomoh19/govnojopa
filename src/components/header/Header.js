import React, { useState } from 'react'
import { NavLink } from "react-router-dom";
import LocaleMenu from '../locale-menu/Locale-menu';
import { useTranslation } from "react-i18next";
import { setCookiesLocale } from "../../utils/cookies";
import './Header.scss'

const Header = ({ mobileMenuHandler, mobileMenu, children }) => {
    const [openLocaleMenu, setOpenLocaleMenu] = useState(false);

    const { t, i18n } = useTranslation()

    const changeLanguage = language => {
        i18n.changeLanguage(language)
        setCookiesLocale(language)
    }

    const toggleLocaleMenu = () => setOpenLocaleMenu(!openLocaleMenu)

    return (
        <>
            <header className="header">
                <NavLink to="/" className="header__logo-wrapper">L.</NavLink>
                {/* DESCTOP MENU */}
                <div className="header__main">
                    <NavLink exact to="/" activeClassName="header__item-active" className="header__item header__main-item">
                        {t('team')}
                    </NavLink>
                    <NavLink to="/befree/follow" activeClassName="header__item-active" className="header__item header__main-item">
                        {t('befree')}
                    </NavLink>
                    <NavLink to="/casting" activeClassName="header__item-active" className="header__item header__main-item">
                        {t('casting')}
                    </NavLink>
                    <NavLink to="/contacts" activeClassName="header__item-active" className="header__item header__main-item">
                        {t('contacts')}
                    </NavLink>
                    <NavLink exact to="/faq" activeClassName="header__item-active" className="header__item header__main-item">
                        FAQ
                    </NavLink>
                </div>
                <div className="locale__wrapper" onClick={toggleLocaleMenu}>
                    <div className="locale__label">
                        {t('locale')}
                        <i style={openLocaleMenu ? { transform: 'rotate(180deg)' } : null} className="fas fa-angle-down icon-arr" />
                    </div>
                    <LocaleMenu setLocale={changeLanguage} openLocaleMenu={openLocaleMenu} />
                </div>

                {/* MOBILE MENU */}
                <div className="header__menu-icon" onClick={mobileMenuHandler}>
                    <span className={mobileMenu ? 'menu-line menu-one menu-line-open-one' : 'menu-line menu-one'} />
                    <span className={mobileMenu ? 'menu-line menu-two menu-line-open-two' : 'menu-line menu-two'} />
                    <span className={mobileMenu ? 'menu-line menu-three menu-line-open-three' : 'menu-line menu-three'} />
                </div>
            </header>
            <main className="main_wrapper">
                {children}
            </main>
        </>
    )
}

export default Header
