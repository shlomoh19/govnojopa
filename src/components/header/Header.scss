@import "../../UI/style.scss";

.header {
  position: relative;
  display: flex;
  justify-content: space-around;
  align-items: center;
  height: 60px;
  background: $lightGrey;
  z-index: 10;
  .header__main {
    display: flex;
    justify-content: space-around;
  }
  .header__item {
    font-size: 20px;
    display: flex;
    justify-content: center;
    align-items: center;
    text-transform: uppercase;
    text-decoration: none;
    color: $black;
    transition: background, transform .2s;
    &:hover {
      color: $blue;
    }
  }
}

.header__item-active {
  color: $blue !important;
}
.header__main-item {
  padding: 0 26px;
}
.locale__label {
  cursor: pointer;
  color: #636e9e;
  font-weight: 400;
}


//MENU

.header__menu-icon {
  position: relative;
  display: none;
  width: 25px;
  height: 20px;
  justify-content: center;
  .menu-line {
    position: absolute;
    background-color: black;
    height: 3px;
    width: 25px;
    transition: all .7s;
  }
  .menu-one {
    top: 0;
  }
  .menu-two {
    top: 47%;
  }
  .menu-three {
    bottom: 0;
  }
}

.menu-line-open-one.menu-one {
  transform: translateY(8.5px) rotate(45deg);
  animation: line-mount-one ease-in-out .7s;
}
.menu-line-open-two.menu-two {
  width: 0;
  animation: menu-line-open-two ease-in-out .5s;
}
.menu-line-open-three.menu-three {
  transform: translateY(-8.5px) rotate(-45deg);
  animation: line-mount-three ease-in-out .7s;
}

@keyframes line-mount-one {
  0% {
    transform: translateY(0px) rotate(0deg);
  }
  100% {
    transform: translateY(8.5px) rotate(400deg);
  }
}

@keyframes line-mount-three {
  0% {
    transform: translateY(0px) rotate(0deg);
  }
  100% {
    transform: translateY(-8.5px) rotate(-400deg);
  }
}

@keyframes menu-line-open-two {
  0% {
    width: 25px;
    opacity: 1;
  }
  100% {
    width: 0;
    transform: translateX(2px);
    opacity: 0;
  }
}

@media (max-width: 1440px) {
  .header {
    font-size: 16px;
    .header__item {
      font-size: 16px;
    }
  }
}

@media (max-width: 1200px) {
  .header {
    font-size: 14px;
  }
}


@media (max-width: 800px) {
  .header {
    padding-left: 20px;
    padding-right: 20px;
    .header__item {
      font-size: 14px;
    }
  }
}

@media (max-width: 560px) {
  .header {
    font-size: 12px;
    .header__item {
      font-size: 12px;
    }
  }
}


@media (max-width: 500px) {
  .header {
    display: flex;
    justify-content: space-between;
    padding: 0px 30px;
    height: 50px;
    .header__item {
      display: none;
    }
    .header__main {
      display: none;
    }
  }
  .header__menu-icon {
    display: flex;
  }
}
