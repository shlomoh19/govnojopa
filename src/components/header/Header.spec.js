import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import Header from './Header'

describe('Header', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Header />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
})