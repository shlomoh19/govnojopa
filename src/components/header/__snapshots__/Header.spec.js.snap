// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Render Header component 1`] = `ShallowWrapper {}`;

exports[`Header renders correctly 1`] = `
<Fragment>
  <header
    className="header"
  >
    <NavLink
      className="header__logo-wrapper"
      to="/"
    >
      L.
    </NavLink>
    <div
      className="header__main"
    >
      <NavLink
        activeClassName="header__item-active"
        className="header__item header__main-item"
        exact={true}
        to="/"
      >
        team
      </NavLink>
      <NavLink
        activeClassName="header__item-active"
        className="header__item header__main-item"
        to="/befree/follow"
      >
        befree
      </NavLink>
      <NavLink
        activeClassName="header__item-active"
        className="header__item header__main-item"
        to="/casting"
      >
        casting
      </NavLink>
      <NavLink
        activeClassName="header__item-active"
        className="header__item header__main-item"
        to="/contacts"
      >
        contacts
      </NavLink>
    </div>
    <div
      className="locale__wrapper"
      onClick={[Function]}
    >
      <div
        className="locale__label"
      >
        locale
        <i
          className="fas fa-angle-down icon-arr"
          style={null}
        />
      </div>
      <LocaleMenu
        openLocaleMenu={false}
        setLocale={[Function]}
      />
    </div>
    <div
      className="header__menu-icon"
    >
      <span
        className="menu-line menu-one"
      />
      <span
        className="menu-line menu-two"
      />
      <span
        className="menu-line menu-three"
      />
    </div>
  </header>
  <main
    className="main_wrapper"
  />
</Fragment>
`;
